src/slider/slider-labels.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import styles from './styles.css.js';
+import { useMobile } from '../internal/hooks/use-mobile/index.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+
+export interface SliderLabelProps {
+  min: number;
+  max: number;
+  referenceValues?: Array<number>;
+  valueFormatter?: (value: number) => string;
+  labelsId: string;
+}
+export default function SliderLabels({ min, max, referenceValues, valueFormatter, labelsId }: SliderLabelProps) {
+  const isMobile = useMobile();
+  const LABEL_THRESHOLD = isMobile ? 3 : 8;
+
+  const getVisibleReferenceValues = () => {
+    if (!referenceValues || referenceValues.length === 0) {
+      return [];
+    }
+
+    const values: number[] = [];
+    let lastValue = 0;
+    for (let i = 0; i <= referenceValues.length; i++) {
+      if (
+        Math.abs(referenceValues[i] - lastValue) >= (max - min) / LABEL_THRESHOLD &&
+        Math.abs(max - referenceValues[i]) >= (max - min) / LABEL_THRESHOLD
+      ) {
+        values.push(referenceValues[i]);
+        lastValue = referenceValues[i];
+      }
+    }
+    return values;
+  };
+
+  function getLabelPosition(index: number) {
+    const minDistance = (max - min) / LABEL_THRESHOLD;
+    const colSpan = Math.floor(minDistance / 2);
```

Comment:
What is colSpan?

Reply:
the grid column span that the label will take up. this is to give the label the most space it can have.